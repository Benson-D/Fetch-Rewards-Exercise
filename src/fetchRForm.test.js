import { render } from '@testing-library/react';
import FetchRewardsForm from './fetchRForm';

const postRewardsData = jest.fn();

const occupations = [
    "testing",
    "testing2"
]

const states = [
    {
        "name": "Alabama",
        "abbreviation": "AL"
    },
    {
        "name": "Illinois",
        "abbreviation": "IL"
    }
]

describe("Fetch Rewards Form", function () {
    it("renders without crashing", function () {
      render(<FetchRewardsForm occupations={occupations} states={states} postFetchReward={postRewardsData}/>);
    }); 

    it("matches snapshot", function () {
        const { container } = render(<FetchRewardsForm occupations={occupations} states={states} postFetchReward={postRewardsData}/>);
        expect(container).toMatchSnapshot();
      });
  });