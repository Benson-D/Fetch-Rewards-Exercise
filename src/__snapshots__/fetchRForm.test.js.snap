// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch Rewards Form matches snapshot 1`] = `
<div>
  <section
    class="col-sm-8 col-md-6 col-lg-4 offset-sm-2 offset-md-3 offset-lg-4"
  >
    <div
      class="FetchRewardsForm card"
    >
      <div
        class="card-body"
      >
        <form>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Name
            </label>
            <input
              class="form-control"
              name="name"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Email
            </label>
            <input
              class="form-control"
              name="email"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Password
            </label>
            <input
              class="form-control"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              Occupation
            </label>
            <select
              class="form-control"
              name="occupation"
            >
              <option>
                testing
              </option>
              <option>
                testing2
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
            >
              State
            </label>
            <select
              class="form-control"
              name="state"
            >
              <option>
                Alabama AL
              </option>
              <option>
                Illinois IL
              </option>
            </select>
          </div>
          <button
            class="btn btn-success"
            disabled=""
          >
             Submit!
          </button>
        </form>
      </div>
    </div>
  </section>
</div>
`;
